plugins/backstage-plugin-coder/src/components/CoderProvider/CoderAuthProvider.tsx

```diff
@@ -55,52 +65,28 @@ export type CoderAuthStatus = AuthState['status'];
 export type CoderAuth = Readonly<
   AuthState & {
     isAuthenticated: boolean;
-    tokenLoadedOnMount: boolean;
     registerNewToken: (newToken: string) => void;
     ejectToken: () => void;
   }
 >;
 
-function isAuthValid(state: AuthState): boolean {
-  return (
-    state.status === 'authenticated' ||
-    state.status === 'distrustedWithGracePeriod'
-  );
-}
-
-type ValidCoderAuth = Extract<
-  CoderAuth,
-  { status: 'authenticated' | 'distrustedWithGracePeriod' }
->;
-
-export function assertValidCoderAuth(
-  auth: CoderAuth,
-): asserts auth is ValidCoderAuth {
-  if (!isAuthValid(auth)) {
-    throw new Error('Coder auth is not valid');
-  }
-}
-
-export const AuthContext = createContext<CoderAuth | null>(null);
+type TrackComponent = (componentInstanceId: string) => () => void;
+export const AuthTrackingContext = createContext<TrackComponent | null>(null);
+export const AuthStateContext = createContext<CoderAuth | null>(null);
 
-export function useCoderAuth(): CoderAuth {
-  const contextValue = useContext(AuthContext);
-  if (contextValue === null) {
-    throw new Error(
-      `Hook ${useCoderAuth.name} is being called outside of CoderProvider`,
-    );
-  }
-
-  return contextValue;
-}
+const validAuthStatuses: readonly CoderAuthStatus[] = [
+  'authenticated',
+  'distrustedWithGracePeriod',
+];
 
-type CoderAuthProviderProps = Readonly<PropsWithChildren<unknown>>;
+function useAuthState(): CoderAuth {
+  const [authToken, setAuthToken] = useState(
+    () => window.localStorage.getItem(TOKEN_STORAGE_KEY) ?? '',
+  );
```

Comment:
Inlined the previous function because it was only used in one spot, and also, with the file getting so big, I wanted as few top-level values hanging around the module as possible